test/behavior/type.zig

```diff
@@ -438,9 +438,9 @@ test "Type.Union" {
         },
     });
     var tagged = Tagged{ .signed = -1 };
-    try testing.expectEqual(Tag.signed, tagged);
+    try testing.expectEqual(Tag.signed, @as(Tag, tagged));
```

Comment:
Fails with
```
lib\std\testing.zig:58:32: error: coercion from enum 'type.test.Type.Union.Tag' to union 'type.test.Type.Union.Tagged' must initialize 'i32' field 'signed'
    return expectEqualInner(T, expected, actual);
                               ^~~~~~~~
test\behavior\type.zig:429:20: note: field 'signed' declared here
    const Tagged = @Type(.{
                   ^~~~~
test\behavior\type.zig:429:20: note: union declared here
    const Tagged = @Type(.{
                   ^~~~~
test\behavior\type.zig:441:28: note: called from here
    try testing.expectEqual(Tag.signed, tagged);
        ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~
```

Reply:
Whilst unions are able to coerce to their tag type, it is not desirable for PTR to be able to trigger such a coercion. Consider this code:
```
const U = union(enum) {
    a: u32,
    b: u32,
};
const x = if (foo) U{ .a = 1 } else .b;
```
It would be very unexpected if the type of `x` here was the tag type of `U`, silently dropping the payload!